app/assets/javascripts/map.js

```diff
@@ -11,19 +12,53 @@
         }
       });
     },
+    destroy: function() {
+      $.each(App.Map.maps, function(_, map) {
```

Comment:
We could simplify this line a little bit, removing the unused parameter and using JavaScript native loops:
```javascript
App.Map.maps.forEach(function(map) {
```